bin/copyTrendAnalysisDataToPrm.sh

```diff
@@ -0,0 +1,247 @@
+# Therefore locking must be done after 
+# * sourcing the file containing the lock function,
+# * sourcing config files,
+# * and parsing commandline arguments,
+# but before doing the actual data trnasfers.
+#
+lockFile="${PRM_ROOT_DIR}/logs/${SCRIPT_NAME}.lock"
+thereShallBeOnlyOne "${lockFile}"
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Successfully got exclusive access to lock file ${lockFile} ..."
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Log files will be written to ${PRM_ROOT_DIR}/logs ..."
+
+#
+# Use multiplexing to reduce the amount of SSH connections created
+# when rsyncing using the group's data manager account.
+# 
+#  1. Become the "${DATA_MANAGER} user who will rsync the data to prm and 
+#  2. Add to ~/.ssh/config:
+#		ControlMaster auto
+#		ControlPath ~/.ssh/tmp/%h_%p_%r
+#		ControlPersist 5m
+#  3. Create ~/.ssh/tmp dir:
+#		mkdir -p -m 700 ~/.ssh/tmp
+#  4. Recursively restrict access to the ~/.ssh dir to allow only the owner/user:
+#		chmod -R go-rwx ~/.ssh
+#
+
+#
+# Get a list of all projects for this group, loop over their run analysis ("run") sub dirs and check if there are any we need to rsync.
+#
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Searching for folders as ${DATA_MANAGER} on ${HOSTNAME_TMP} in ${CHRONQC_REPORTS_DIRS}/*"
+# shellcheck disable=SC2029
+mapfile -t chronQCDateFolders < <(ssh "${DATA_MANAGER}"@"${HOSTNAME_TMP}" \"find "${CHRONQC_REPORTS_DIRS}\" -maxdepth 1 -mindepth 1 -type d")
+if [[ "${#chronQCDateFolders[@]}" -eq '0' ]]
+then
+	log4Bash 'WARN' "${LINENO}" "${FUNCNAME:-main}" '0' "No folders found @ ${CHRONQC_REPORTS_DIRS}."
+else
+	for chronQCDateFolder in "${chronQCDateFolders[@]}"
+	do
+		chronQCDateFolderName=$(basename "${chronQCDateFolder}")
+		controlFileBase="${PRM_ROOT_DIR}/logs/trendanalysis/${chronQCDateFolderName}"
+		export JOB_CONTROLE_FILE_BASE="${controlFileBase}.${SCRIPT_NAME}"
+		
+		if [[ -e "${JOB_CONTROLE_FILE_BASE}.finished" ]]
+		then
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Skipping already processed batch ${chronQCDateFolderName}."
+		else
+			log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Processing chronQCDateFolder ${chronQCDateFolderName} ..."
+			mkdir -p "${PRM_ROOT_DIR}/logs/trendanalysis/"
+			touch "${JOB_CONTROLE_FILE_BASE}.started"
+			rsync -av --progress --log-file="${JOB_CONTROLE_FILE_BASE}.started" --chmod='Du=rwx,Dg=rsx,Fu=rw,Fg=r,o-rwx' "${dryrun:---progress}" \
+				"${DATA_MANAGER}@${HOSTNAME_TMP}:${chronQCDateFolder}" \
+				"${PRM_ROOT_DIR}/trendanalysis/" \
+			|| {
+				mv "${JOB_CONTROLE_FILE_BASE}."{started,failed} 
+				log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" "${?}" "Failed to rsync ${DATA_MANAGER}@${HOSTNAME_TMP}:${chronQCDateFolder} dir. See ${JOB_CONTROLE_FILE_BASE}.failed for details."
+			}
+	
+			mv "${JOB_CONTROLE_FILE_BASE}."{started,finished}
+		fi
+	done
+fi
+
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' 'Finished successfully!'
```

Comment:
ook als de rsync vaalt krijg je een toch een 'Finished successfully', misschien hier een 'script finished'. en na de mv van de log file van .started naar .fished een 'finished successfullt'?

Reply:
uhm, die loop gaat per folder. Dat ie finished succesfully zegt is het script zelf, die heeft in principe geen .started/.failed/.finished, we krijgen alleen logfiles per folder/project